lab-anthony/route/note-route.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+const Note = require('../model/note');
+const response = require('../lib/response');
+const Storage = require('../lib/storage');
+
+const dataDir = new Storage(`${__dirname}/../data`);
+
+var notePool = {};
+
+module.exports = function(router) {
+  router
+    .post('/api/note', function(req, res) {
+      if (!req.body || !req.body.content) {
+        return response(400, 'bad request')(res);
+      }
+
+      const note = new Note(req.body.content);
+      notePool[note.id] = note;
+      console.log(note);
+      dataDir.setItem('notes', note).then(function(item){
+        console.log('the note is ', item);
+        return response(200, item)(res);
+      }).catch(function(err){
+        console.log(err);
+        return response(400, 'bad request')(res);
+      });
+    })
+    .get('/api/note', function(req, res) {
+      const noteId = req.url.query.id;
```

Comment:
This is a little dangerous--your code will break if they don't pass you a query string.